src/deluge/gui/views/arranger_view.cpp

```diff
@@ -3290,3 +3309,40 @@ void ArrangerView::requestRendering(UI* ui, uint32_t whichMainRows, uint32_t whi
 		uiNeedsRendering(ui, whichMainRows, whichSideRows);
 	}
 }
+
+void ArrangerView::drawClipName(Output* output, Clip* clip) {
+
+	deluge::hid::display::oled_canvas::Canvas& canvas = hid::display::OLED::main;
+
+	int32_t yPos = OLED_MAIN_TOPMOST_PIXEL + 3;
+	int32_t textSpacingX = kTextTitleSpacingX;
+	int32_t textSpacingY = kTextTitleSizeY;
+
+	canvas.drawStringCentred("ARRANGER", yPos, kTextSpacingX, kTextSpacingY);
+
+	if (output->type == OutputType::AUDIO) {
+		AudioClip* audioclip = (AudioClip*)clip;
+		clip->clipName.set(&audioclip->sampleHolder.filePath);
```

Comment:
this shouldn't be handled by the view, it should be set when the audio clip's file path is set or recording is finalized

Reply:
Going to change this part and look where the entry point is ( maybe on a few different places )